dashboard/app/controllers/vocabularies_controller.rb

```diff
@@ -36,11 +36,25 @@ def update
     end
   end
 
+  # GET /courses/:course_name/vocab/edit
+  def edit
+    @course_version = find_matching_course_version
+    @vocabularies = @course_version.vocabularies.order(:word)
+  end
+
   private
 
   def vocabulary_params
     vp = params.transform_keys(&:underscore)
     vp = vp.permit(:id, :key, :word, :definition, :course_version_id)
     vp
   end
+
+  def find_matching_course_version
+    matching_unit_group = UnitGroup.find_by_name(params[:course_name])
+    return matching_unit_group.course_version if matching_unit_group
+    matching_standalone_course = Script.find_by_name(params[:course_name])
+    return matching_standalone_course&.get_course_version if matching_standalone_course&.is_course
+    return nil
+  end
```

Comment:
The above method looks like it might be some of the mess you were talking about as it relates to course version. I don't see a way around this in the short term. in the future, we want to move properties like "name" out of the Script and UnitGroup and into the CourseVersion. Then, this kind of operation should be easier to perform. 

Does any of this seem a bit more sane knowing that "name" should eventually get moved into CourseVersion?

Reply:
Yes that would be so much easier here! That's essentially what this function is looking for, except that we can't know which model holds the name until we look for it

Reply:
Furthermore -- one of the reasons I was considering not having this logic in `vocabularies_controller` was because this helper function would need to be shared/duplicated when we write the All Resources page. I figured I could kick that can down the road and move it when it actually needs to be shared, but having a simpler way to query for a course would make this issue largely moot.

Reply:
Good point -- this doesn't really belong in the vocabularies controller. when it comes time to build the All Resources page, I think it would make sense to move this into the CourseVersion model. I don't think we'll get to moving the name out of Script and UnitGroup into CourseVersion until "Q2 Tech Debt".